samples/switchless/host/host.c

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "switchless_u.h"
+
+void host_helloworld()
+{
+    fprintf(stdout, "Hello world from regular OCALL\n");
+}
+
+void host_helloworld_switchless()
+{
+    fprintf(stdout, "Hello world from switchless OCALL\n");
+}
+
+static bool check_simulate_opt(int* argc, const char* argv[])
+{
+    for (int i = 0; i < *argc; i++)
+    {
+        if (strcmp(argv[i], "--simulate") == 0)
+        {
+            fprintf(stderr, "Running in simulation mode\n");
+            memmove(&argv[i], &argv[i + 1], (*argc - i) * sizeof(char*));
+            (*argc)--;
+            return true;
+        }
+    }
+    return false;
+}
+
+int main(int argc, const char* argv[])
+{
+    oe_enclave_t* enclave = NULL;
+    oe_result_t result;
+    int ret = 1;
+
+    if (argc != 2 && argc != 3)
+    {
+        fprintf(stderr, "Usage: %s ENCLAVE_PATH [--simulate]\n", argv[0]);
+        return 1;
+    }
+
+    uint32_t flags = OE_ENCLAVE_FLAG_DEBUG;
+    if (check_simulate_opt(&argc, argv))
+    {
+        flags |= OE_ENCLAVE_FLAG_SIMULATE;
+    }
+
+    // Enable switchless and configure host worker number
+    oe_enclave_config_context_switchless_t config = {2, 0};
+    oe_enclave_config_t configs[] = {{
```

Comment:
Is there any recommendation on how to configure (number threads) to get best performance from this switchless feature?

Reply:
Added recommendations in `How does OE support switchless OCALLs`

Reply:
Thanks for addressing my comment